File: Data Structures & Algorithms/binary-tree-right-side-view/submission-11.js
/**
 * Definition for a binary tree node.
 * class TreeNode {
 *     constructor(val = 0, left = null, right = null) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

class Solution {
    /**
     * @param {TreeNode} root
     * @return {number[]}
     */
    rightSideView(root) {
        let queue = new Queue();
        let res = [];

        if(!root) return [];

        queue.enqueue(root);

        while(!queue.isEmpty()){

            let len = queue.size();
            let first = true;
            while(len > 0){
                let node = queue.dequeue();
                if(first) res.push(node.val);
                first = false;

                if(node.right) queue.enqueue(node.right);
                if(node.left) queue.enqueue(node.left);

                len--;
            }
        }
        return res;
    }
}

File: Data Structures & Algorithms/insert-into-a-binary-search-tree/submission-11.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def insertIntoBST(self, root: Optional[TreeNode], val: int) -> Optional[TreeNode]:
        
        if not root:
            return TreeNode(val)

        point = root

        while point:
            
            if val > point.val:
                if not point.right:
                    break
                point = point.right
            else:
                if not point.left:
                    break
                point = point.left

        if val > point.val:
                point.right = TreeNode(val)
        else:
                point.left = TreeNode(val)


        return root